### ISSUE TITLE
Java to CPP translation : vector -- remove(Object o)

### ISSUE THREAD

**Issue by lublud** [darrivau/GOOL#3]
When removing from a list an item in Java:
```java
list.remove("toto");
```
The translation in C++ is the following:
```cpp
for(std::vector<int>::size_type i = 0; i != list->size(); i++)
{
    if (boost::any_cast<std::string*>(list->at(i)) == ( new std::string ( "toto" ) ) )
    {
        list->erase(list->begin()+i);
        break;
    }
};  
```
Though, this instruction:
```cpp
boost::any_cast<std::string*>(list->at(i))
```
returns the address of the i-th item instead of its content.
* expected toto
* result 0x22e7080 // or other address

**Comment by darrivau** [comment]
Note that in standard C++, iteration within a list, vector or any containers should be done with iterators like :

// find example
#include <iostream>     // std::cout
#include <algorithm>    // std::find
#include <vector>       // std::vector

int main () {
  // using std::find with array and pointer:
  int myints[] = { 10, 20, 30, 40 };
  int * p;

  p = std::find (myints, myints+4, 30);
  if (p != myints+4)
    std::cout << "Element found in myints: " << *p << '\n';
  else
    std::cout << "Element not found in myints\n";

  // using std::find with vector and iterator:
  std::vector<int> myvector (myints,myints+4);
  std::vector<int>::iterator it;

  it = find (myvector.begin(), myvector.end(), 30);
  if (it != myvector.end())
    std::cout << "Element found in myvector: " << *it << '\n';
  else
    std::cout << "Element not found in myvector\n";

  return 0;
}

**Comment by darrivau** [comment]
Corrected in commit 887db7d with `list->erase(std::find(list->begin(),list->end(),"toto"))`
